### ISSUE TITLE
company-doc-buffer is used even if company is not installed

### ISSUE THREAD

**Issue by minad** [emacs-lsp/lsp-mode#3225]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [x] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [x] You may also try reproduce the issue using clean environment using the following command
```bash
emacs -q -l lsp-start-plain.el
```
where `lsp-start-plain.el` can be downloaded [here](https://github.com/emacs-lsp/lsp-mode/blob/master/scripts/lsp-start-plain.el).

### Bug description

Company uses company-doc-buffer even if Company is not used as completion framework. Lsp-mode should instead provide its own implementation. This issue has been observed in my completion UI Corfu, see https://github.com/minad/corfu/issues/87.

See https://github.com/emacs-lsp/lsp-mode/blob/1ece3b81ad10d1b10a2c73c6617b4426e673cbc7/lsp-completion.el#L520-L521

### Steps to reproduce

Hitting M-h in Corfu on a candidate shows the doc buffer. This calls :company-doc-buffer on the side of lsp-mode, which is ill-defined.

### Expected behavior

Lsp-mode should provide a doc-buffer, it should not rely on a Company function.

### Which Language Server did you use?

Unknown.

### OS

Linux

### Error callstack

```emacs-lisp
Debugger entered--Lisp error: (void-function company-doc-buffer)
  company-doc-buffer(nil)
  #f(compiled-function (&rest args) #<bytecode -0xb6c74c414fddab3>)(#("atomic=" 0 7 (lsp-completion-resolved t lsp-completion-item #<hash-table equal 5/5 0x1ff0c67bcd93> lsp-sort-text "04.9999.atomic" lsp-completion-start-point 3304 lsp-completion-markers (3304 #<marker (moves after insertion) at 3307 in linkerd.py>) lsp-completion-prefix "ato" match-data (0 3 0 1 1 2 2 3))))
  corfu-show-documentation()
  funcall-interactively(corfu-show-documentation)
  command-execute(corfu-show-documentation)
```

### Anything else?

Original issue by @antifuchs in https://github.com/minad/corfu/issues/87.

**Comment by minad** [comment]
As pointed out by @antifuchs, the fix for this issue is simple. Instead of using `company-doc-buffer`, lsp-mode should define its own function. The function is easy enough to copy verbatim to lsp-mode.

https://github.com/company-mode/company-mode/blob/6abb232acde15e1a7bf402a57f7bb982edf2de7d/company.el#L2737-L2745

**Comment by minad** [comment]
> Closed #3225 <https://github.com/emacs-lsp/lsp-mode/issues/3225> via
> #3229 <https://github.com/emacs-lsp/lsp-mode/pull/3229>.

Thank you!
